File: quadtrees/dev/sector.py
# SECTOR BOUNDARY CLASS

from point import Point

class Sector:
	'''
	'''
	def __init__(self, center: object, half_dim: float) -> None:
		self.center = center
		self.half_dim = half_dim

	def contains_point(self, point: object) -> bool:
		pass

	def intersects(self, boundary: object) -> bool:
		pass


if __name__ == "__main__":
	pass

File: quadtrees/dev/cpp/sector.cpp
// SECTOR (BOUNDARY) SOURCE FILE

#include "sector.h"

Sector::Sector(Point cen, int rad) : center{cen}, radius{rad} {}

bool Sector::contains_point(Point point)
{
	// with overloads:
	return (point < (center + radius)) && (point > (center - radius)) ? true : false;

	// x_cond = (point.x < (center.x + radius)) && (point.x > (center.x - radius));
	// y cond = (point.y < (center.y + radius)) && (point.y > (center.y - radius));
	// return (x_cond && y_cond) ? true : false;
}